corehq/apps/app_manager/models.py

```diff
@@ -3507,6 +3498,31 @@ def get_filter_value(self, user, ui_filter):
         return None
 
 
+class AncestorLocationTypeFilter(ReportAppFilter):
+    ancestor_location_type_name = StringProperty()
+
+    def get_filter_value(self, user, ui_filter):
+        """
+        A list of locations who have the same ancestor as user param
+
+        :raises LocationType.DoesNotExist: if ancestor_location_type_name does not match a location type
+        """
+        from corehq.apps.locations.models import LocationType
+        from corehq.apps.reports_core.filters import Choice
+
+        ancestor_type = LocationType.objects.get(name=self.ancestor_location_type_name)
+        location = user.sql_location
+        if not location:
+            return []
+        try:
+            ancestor = [a for a in location.get_ancestors() if a.location_type == ancestor_type][0]
+        except KeyError:
+            # No location of that type among ancestors
+            return []
+        return [Choice(value=l.location_id, display=None) for l in ancestor.get_descendants(include_self=True)]
```

Comment:
why are the descendants being returned? I would have thought you wanted to return the `ancestor` itself (or its id)